src/query_plan/operation.rs

```diff
@@ -68,6 +70,61 @@ pub struct NormalizedOperation {
     pub(crate) fragments: Arc<IndexMap<Name, Node<NormalizedFragment>>>,
 }
 
+pub(crate) struct NormalizedDefer {
+    pub operation: NormalizedOperation,
+    pub has_defers: bool,
+    pub assigned_defer_labels: HashSet<NodeStr>,
+    pub defer_conditions: IndexMap<String, IndexSet<String>>,
+}
+
+impl NormalizedOperation {
```

Comment:
Aside from the `with_normalized_defer` and `without_defer` stubs, all changes in this file are to make the selection set merging methods iterate over references instead of over owned instances.